### DISCUSSION TITLE
JSON to YAML conversion - option for explicit string quoting

### DISCUSSION THREAD

**Discussion by harrel56** [mikefarah/yq#1938] [Q&A] [answered]
Hey community,

I'm using `yq -Poy file.json` for conversion to YAML and for example:
```js
{
  "date": "12:00:00"
}
```
is parsed to:
```yml
date: 12:00:00
```

Is there any option to explicitly quote all string values? So the output would become:
```yaml
date: '12:00:00' # or "12:00:00"
```

TBH I don't know if parsing to this form wouldn't cause some troubles with escaping?

And this is causing issues for me as unquoted `12:00:00` is being parsed (java, snakeyaml) as a number (`43200`, number of seconds if I'm not wrong). 

Thanks in advance for any help on this!

**Comment by mikefarah** [comment, marked as answer]
Yep, you can give it a style expression to put values in double quotes, see https://mikefarah.gitbook.io/yq/operators/env-variable-operators#envsubst-options for details.

```
yq -Poy '.date style="double"'  file.json 
```

**Reply by harrel56** [reply]
TBH I cannot make it work directly when converting between JSON and YAML, I have to do this in 2 separate steps:
```sh
yq -Poy file.json | yq '.date style="single"'
```
Also, it'd be great if I could set style on conversion operation and apply the quotations only to actual JSON strings. I see there is a `kind` operator but it looks like it's not able to distinguish strings apart from e.g. numbers. Probably I would imagine something like this:
```sh
yq -Poy '.. | select(kind == "string") style="single"' file.json 
```

But still the 2 step approach works for my case, so it's all fine. Thank you for your help!
